### PR TITLE
Harden OpenSSL crypto policy

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- add rule removing CCM ciphersuite from OpenSSL crypto policy

#### Rationale:

- meeting CC requirements

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule harden_openssl_crypto_policy:
 Bash remediation is newly added.
 The rule doesn't occur in any profile nor product.
 Ansible remediation newly added.
 OVAL check is newly added.

File: linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/rule.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    are configured e.g. cipher suites. Currently particular requirements
+    specified by CC are stricter compared to any existing Crypto Policy.
+
+severity: medium
+
+identifiers:
+    cce@rhel8: 84286-4
+    cce@rhcos4: 84285-6
+
+references:
```
Comment (vojtapolasek) [inline linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/rule.yml:32]: @redhatrises  Hello, could you please verify if these references for nist and stig are correct? Thank you.

File: linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/tests/incorrect_followed_by_correct.pass.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+# platform = multi_platform_fedora,Red Hat Enterprise Linux 8
+# profiles = xccdf_org.ssgproject.content_profile_ospp
```
Comment (jan-cerny) [inline linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/tests/incorrect_followed_by_correct.pass.sh:3]: Will you add the rule `harden_openssl_crypto_policy` to the ospp profile? At this moment, the rule isn't part of any profile.

This causes the test suite to shout.

```
[jcerny@thinkpad tests{pr/5925}]$ python3 test_suite.py rule --libvirt qemu:///system ssgts_rhel8 --datastream ../build/ssg-rhel8-ds.xml harden_openssl_crypto_policy
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/tests/logs/rule-custom-2020-07-14-1354/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy
ERROR - Script incorrect_followed_by_correct.pass.sh using profile xccdf_org.ssgproject.content_profile_ospp found issue:
ERROR - Rule xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy has not been evaluated! Wrong profile selected in test scenario?
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy'.
ERROR - Script empty_policy.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp found issue:
ERROR - Rule xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy has not been evaluated! Wrong profile selected in test scenario?
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy'.
ERROR - Script correct_followed_by_incorrect.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp found issue:
ERROR - Rule xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy has not been evaluated! Wrong profile selected in test scenario?
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy'.
ERROR - Script correct.pass.sh using profile xccdf_org.ssgproject.content_profile_ospp found issue:
ERROR - Rule xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy has not been evaluated! Wrong profile selected in test scenario?
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy'.
ERROR - Script missing_file.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp found issue:
ERROR - Rule xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy has not been evaluated! Wrong profile selected in test scenario?
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy'.
ERROR - Script correct_commented.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp found issue:
ERROR - Rule xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy has not been evaluated! Wrong profile selected in test scenario?
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy'.
ERROR - Script incorrect_policy.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp found issue:
ERROR - Rule xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy has not been evaluated! Wrong profile selected in test scenario?
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_harden_openssl_crypto_policy'.
```

Also the profiles comment in test scenarios  header isn't necessary when the rule isn't parametrized by an XCCDF value.

File: linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/rule.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    nist: AC-17(a),AC-17(2),CM-6(a),MA-4(6),SC-13,SC-12(2),SC-12(3)
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/rule.yml:33]: @vojtapolasek these should be: `SC-8(1),sc-13`

File: linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/rule.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    ospp: FCS_TLSC_EXT.1.1
+    srg: SRG-OS-000250-GPOS-00093
```
Comment (redhatrises) [inline linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/rule.yml:35]: This should be: `SRG-OS-000396-GPOS-00176,SRG-OS-000424-GPOS-00188,SRG-OS-000478-GPOS-00223`

**Comment by mildas** [comment]
**Changes identified:**
Rule harden_openssl_crypto_policy:
 Ansible remediation newly added.
 OVAL check is newly added.
 The rule doesn't occur in any profile nor product.
 Bash remediation is newly added.

File: linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/tests/incorrect_followed_by_correct.pass.sh (hunk 2 above)
Comment (vojtapolasek) [reply linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/tests/incorrect_followed_by_correct.pass.sh:3]: No, the rule is here to fullfill specific requirement and it won't be placed into upstream profile.

**Comment by mildas** [comment]
**Changes identified:**
Rule harden_openssl_crypto_policy:
 The rule doesn't occur in any profile nor product.
 Ansible remediation newly added.
 Bash remediation is newly added.
 OVAL check is newly added.

File: linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/rule.yml (hunk 4 above)
Comment (vojtapolasek) [reply linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/rule.yml]: Thank you for assistance.

File: linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/tests/incorrect_followed_by_correct.pass.sh (hunk 2 above)
Comment (jan-cerny) [reply linux_os/guide/system/software/integrity/crypto/harden_openssl_crypto_policy/tests/incorrect_followed_by_correct.pass.sh:3]: @vojtapolasek OK, then why do you put `profiles` to the test scenario header?

**Comment by vojtapolasek** [comment]
Because the rule is specific to OSPP profile. I use the (all) profile while testing it. I know it is not ideal. But similar rules such as harden_ssh_client_crypto_policy do it in the same way.

**Comment by jan-cerny** [comment]
@vojtapolasek The purpose of the `# profiles` metadata in the test scenarios header is to mark test scenarios that are giving desired test results only if the test scearios are run using the given profile and if they are run using a different profile they give inaccurate test results. That usually happens when variables are used by the rule and the test scenarios are testing a specific value selected in the given profile. The test scenarios in this PR are not the case, the test scenarios will give the same test results if the rule is selected in any profile.

As this rule isn't expected to be selected in anything else than OSPP it's kind of special so I won't block the PR because of it, but I think that an unparseable comment instead of it could help you more.

**Comment by vojtapolasek** [comment]
I understand, thank you for the explanation. I will consider it in upcoming PRs.

**Pull request merged**